File: DSA/RATNA_KHANNA_590013141/DAY_19_QUESTION_1.c
#include <stdio.h>
int search(int arr[], int size, int target) 
{
    int left = 0, right = size - 1;
    while (left <= right) 
    {
        int mid = left + (right - left) / 2;
        if (arr[mid] == target)
            return mid;
        if (arr[left] <= arr[mid]) 
        {
            if (target >= arr[left] && target < arr[mid])
                right = mid - 1;
            else
                left = mid + 1;
        }
        else 
        {
            if (target > arr[mid] && target <= arr[right])
                left = mid + 1;
            else
                right = mid - 1;
        }
    }
    return -1; 
}
int main() 
{
    int n, target;
    printf("Enter the size of the array: ");
    scanf("%d", &n);
    int arr[n];
    printf("Enter the elements of the rotated sorted array:\n");
    for (int i = 0; i < n; i++) 
    {
        scanf("%d", &arr[i]);
    }
    printf("Enter the target value to search: ");
    scanf("%d", &target);
    int result = search(arr, n, target);
    if (result != -1)
        printf("Target found at index: %d\n", result);
    else
        printf("Target not found in the array.\n");

    return 0;
}

File: DSA/SakshamKashyap_590014183/day19/day19_Question_1.py
# function to rotate the array
def array_rotate(arr,k):
    n = len(arr)
    k = k % n 
    return arr[-k:] + arr[:-k]

def binary_search(arr,value):
    left = 0
    right = len(arr)
    while left <= right:
        mid = (left+right)//2
        if arr[mid] == value:
            return mid
        elif arr[mid] < value:
            left = mid + 1
        else:
            right = mid - 1
    return -1

#prompt the user to enter the array

My_array = list(map(int,input("Enter the element of array(Give space between them): ").strip().split()))

pivot_point = int(input("Enter the step u want to rotate the array: "))

My_array.sort()

print("My array is: ",My_array)

rotated_array = array_rotate(My_array,pivot_point)

print("Rotated array: ",rotated_array)

target_value = int(input("Enter the value you want to find in array: "))

result = binary_search(rotated_array,target_value)

if result != -1:
    print("The index of finding elements is: ",result)
else:
    print(result)

